pgrx/src/array/flat_array.rs

```diff
@@ -104,7 +106,7 @@ where
 
         let ndims = N;
         if N == 0 {
-            return Ok(FlatArray::new_empty(memcx));
+            return FlatArray::new_empty(memcx);
```

Comment:
Making code like this simpler, even if it's subtle right now, seems like it will be an increasingly bigger win down the line.